src/Adapter/InMemoryAdapter.php

```diff
@@ -62,34 +57,74 @@ public function appendTo(StreamName $streamName, Iterator $domainEvents): void
         $this->streams[$streamName->toString()] = $appendIterator;
     }
 
-    public function load(StreamName $streamName, int $minVersion = null): ?Stream
-    {
+    public function load(
+        StreamName $streamName,
+        int $fromNumber = 0,
+        int $count = null
+    ): ?Stream {
         if (! isset($this->streams[$streamName->toString()])) {
             return null;
         }
 
         $streamEvents = $this->streams[$streamName->toString()];
 
-        if (null !== $minVersion) {
-            $filteredEvents = [];
+        $filteredEvents = [];
+
+        foreach ($streamEvents as $streamEvent) {
+            if ($streamEvent->version() >= $fromNumber) {
+                $filteredEvents[] = $streamEvent;
+            }
+        }
 
-            foreach ($streamEvents as $streamEvent) {
-                if ($streamEvent->version() >= $minVersion) {
-                    $filteredEvents[] = $streamEvent;
+        if (null !== $count) {
+            foreach ($filteredEvents as $key => $streamEvent) {
+                if ($streamEvent->version() > ($fromNumber + $count - 1)) {
```

Comment:
Is a stream always ordered by version? Maybe we could 'break' the previous loop early instead of set/unset.